### PR TITLE
EVG-14332 add task failed or blocked notification type

### PR DESCRIPTION
**Pull request by ablack12**
[EVG-14332](https://jira.mongodb.org/browse/EVG-14332)

### Description 
Notify if the task is failed or blocked. (Could've made this just notify on blocked; I thought a bit about the reporter's use case / general use case for this and I think failed + blocked does make the most sense, but the work to add "just blocked" would be similar to this if it's ever requested in the future.)

### Screenshots
image

### Testing 
Tested on staging with blocked and failed task (pictures available on Evergreen PR).

### Evergreen PR 
https://github.com/evergreen-ci/evergreen/pull/4914

### CODE REVIEW

**Comment by ablack12** [comment]
@ziyielenachen I'm sorry for this commit 🙈

File: src/pages/task/actionButtons/TaskNotificationModal.tsx [hunk 1]
```diff
@@ -86,6 +86,12 @@ export const triggers: Trigger[] = [
     resourceType: "TASK",
     payloadResourceIdKey: "id",
   },
+  {
+    trigger: "task-failed-or-blocked",
+    label: "this task fails or is blocked",
```
Comment (arjunpatel100) [inline src/pages/task/actionButtons/TaskNotificationModal.tsx:91]: nit: can we capitalize the first letter to match the other labels

**Review by arjunpatel100 — APPROVED** [review]

File: src/pages/task/actionButtons/TaskNotificationModal.tsx (hunk 1 above)
Comment (ablack12) [reply src/pages/task/actionButtons/TaskNotificationModal.tsx:91]: I'm so glad you caught this it would've upset me so much

**Comment by ziyielenachen** [comment]
:rolling_on_the_floor_laughing::rolling_on_the_floor_laughing:

**Review by khelif96 — APPROVED** [review]

**Comment by khelif96** [comment]
Don't forget to update the tests.

**Review by malikchaya2 — APPROVED** [review]

**Pull request merged**